beacon-chain/cache/sync_committee_head_state.go

```diff
@@ -47,9 +51,13 @@ func (c *SyncCommitteeHeadStateCache) Get(slot types.Slot) (state.BeaconState, e
 	if !exists {
 		return nil, ErrNotFound
 	}
-	st, ok := val.(*stateAltair.BeaconState)
+	var st state.BeaconState
```

Comment:
same thing here, check if phase 0 and return incorrect type

Reply:
For this, we need to convert the interface. Would you prefer something like the following?

```go
	_, ok := st.(*v1.BeaconState)
	if ok {
		return ErrIncorrectType
	}

	var st state.BeaconState
	st, ok := val.(*v2.BeaconState)
	if !ok {
		st, ok = val.(*v3.BeaconState)
		if !ok {
			return nil, ErrIncorrectType
		}
	}
	return st, nil
```